main/src/com/google/refine/commands/lang/LoadLanguageCommand.java

```diff
@@ -113,32 +130,46 @@ public void doPost(HttpServletRequest request, HttpServletResponse response)
                 ObjectNode node = ParsingUtilities.mapper.createObjectNode();
                 node.set("dictionary", translations);
                 node.set("lang", new TextNode(bestLang));
-                respondJSON(response, node);
+                Command.respondJSON(response, node);
             } catch (IOException e) {
                 logger.error("Error writing language labels to response stream");
+                Command.respondException(response, e);
             }
         } else {
             logger.error("Failed to load any language files");
+            Command.respondException(response, new IOException("No language files"));
         }
     }
 
-    static ObjectNode loadLanguage(RefineServlet servlet, String modname, String lang) throws UnsupportedEncodingException {
+    static ObjectNode loadLanguage(RefineServlet servlet, String strModule, String strLang)
+            throws UnsupportedEncodingException {
+        ButterflyModule module = servlet.getModule(strModule);
+        String strLangFile = "translation-" + strLang + ".json";
+        String strMessage = "[" + strModule + ":" + strLangFile + "]";
+        File langFile = new File(module.getPath(), "langs" + File.separator + strLangFile);
+        FileInputStream fisLang = null;
 
-        ButterflyModule module = servlet.getModule(modname);
-        File langFile = new File(module.getPath(), "langs" + File.separator + "translation-" + lang + ".json");
         try {
-            Reader reader = new BufferedReader(new InputStreamReader(new FileInputStream(langFile), "UTF-8"));
-            return ParsingUtilities.mapper.readValue(reader, ObjectNode.class);
-        } catch (FileNotFoundException e1) {
+            fisLang = new FileInputStream(langFile);
+        } catch (FileNotFoundException e) {
             // Could be normal if we've got a list of languages as fallbacks
-        } catch (IOException e) {
-            logger.error("JSON error reading/writing language file: " + langFile, e);
+            logger.warn("Language file " + strMessage + " not found", e);
```

Comment:
Perhaps an `info` level would be more appropriate here as it is an expected scenario: for instance, if I use OpenRefine in Japanese and some extension has not been translated to Japanese, it is normal to fall back on English for that extension.

Reply:
Yeah, I'm onboard with that.